jplag.endToEndTesting/src/main/java/de/jplag/end_to_end_testing/helper/JPlagTestSuiteHelper.java

```diff
@@ -64,8 +75,103 @@ public TestCaseModel createNewTestCase(String[] classNames, String functionName)
+        logger.debug("Class instance was cleaned!");
+    }
+
+    /**
+     * @return Path to the stored test results
+     */
+    public Path getResultJsonPath() {
+        return resultJsonPath;
+    }
+
+    /**
+     * @return all resources names for the current specified language
+     */
+    public String[] getAllTestFileNames() {
+        return resourceNames;
+    }
+
+    /**
+     * saves all temporary results for a test into a json file
+     * @param jplagComparisonList list of elements to be saved
+     * @throws StreamWriteException Intermediate base class for all read-side streaming processing problems,
+     * includingparsing and input value coercion problems.
+     * @throws DatabindException Intermediate base class for all databind level processing problems, asdistinct from
+     * stream-level problems or I/O issues below.
+     * @throws IOException Signals that an I/O exception of some sort has occurred. Thisclass is the general class of
+     * exceptions produced by failed orinterrupted I/O operations.
+     * @throws NoSuchAlgorithmException when no hash algorithm could be found
+     */
+    public void saveTemporaryResult(List<JPlagComparison> jplagComparisonList, String functionName)
+            throws StreamWriteException, DatabindException, IOException, NoSuchAlgorithmException {
+        for (JPlagComparison jplagComparison : jplagComparisonList) {
+            JsonHelper.writeResultModelToJsonFile(new ResultModel(jplagComparison, getTestHashCode(jplagComparison)),
+                    LanguageToPathMapper.getTemporaryResultPathFromLanguageOption(languageOption).toString(), functionName,
+                    getTemporaryFileNameForJson(jplagComparison));
+        }
+    }
+
+    /**
+     * Creates a unique hash from the submissions in the JPlagComparison object which is used to find the results in the
+     * json files.
+     * @param jPlagComparison object from which the hash should be generated
+     * @return unique identifier for test case recognition
+     * @throws NoSuchAlgorithmException when no hash algorithm could be found
+     */
+    public String getTestHashCode(JPlagComparison jPlagComparison) throws NoSuchAlgorithmException {
+        String testFileNamesInFirstSubmission = new String();
+        String testFileNamesInSecondSubmission = new String();
+        for (File file : jPlagComparison.getFirstSubmission().getFiles()) {
+            testFileNamesInFirstSubmission += file.getName().toString();
+        }
+        for (File file : jPlagComparison.getSecondSubmission().getFiles()) {
+            testFileNamesInSecondSubmission += file.getName().toString();
+        }
+        MessageDigest digest = MessageDigest.getInstance("SHA-256");
+        byte[] encodedhash = digest.digest((testFileNamesInFirstSubmission + testFileNamesInSecondSubmission).getBytes(StandardCharsets.UTF_8));
+
+        StringBuilder stringBuilder = new StringBuilder();
+        for (byte nextByte : encodedhash) {
```

Comment:
new String(byte[], ..)  or similar should exist

Reply:
was removed